hudi-utilities/src/test/java/org/apache/hudi/utilities/functional/TestHoodieDeltaStreamer.java

```diff
@@ -1695,6 +1695,54 @@ public void testJdbcSourceIncrementalFetchInContinuousMode() {
     }
   }
 
+  @Test
+  public void testInsertOverwrite() throws Exception {
+    String tableBasePath = dfsBasePath + "/insert_overwrite";
+    // Initial insert
+    HoodieDeltaStreamer.Config cfg = TestHelpers.makeConfig(tableBasePath, WriteOperationType.INSERT);
+    new HoodieDeltaStreamer(cfg, jsc).sync();
+    TestHelpers.assertRecordCount(1000, tableBasePath + "/*/*.parquet", sqlContext);
+    TestHelpers.assertDistanceCount(1000, tableBasePath + "/*/*.parquet", sqlContext);
+    TestHelpers.assertCommitMetadata("00000", tableBasePath, dfs, 1);
+    // No new data => no commits.
+    cfg.sourceLimit = 0;
+    new HoodieDeltaStreamer(cfg, jsc).sync();
+    TestHelpers.assertRecordCount(1000, tableBasePath + "/*/*.parquet", sqlContext);
+    TestHelpers.assertDistanceCount(1000, tableBasePath + "/*/*.parquet", sqlContext);
+    TestHelpers.assertCommitMetadata("00000", tableBasePath, dfs, 1);
+    // insert overwrite
+    cfg.sourceLimit = 1000;
+    cfg.operation = WriteOperationType.INSERT_OVERWRITE;
```

Comment:
- I see lot of commonality between both these tests. Can we reuse code as much as possible. 
- InsertOverwrite overwrites only the partitions matching the incoming records. But guess we are not really testing that. Do you think we can test this scenario.

Reply:
hi @nsivabalan ,
for the 2nd point 
do we need to cover the test case here, because I see test (https://github.com/apache/hudi/blob/master/hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/client/TestHoodieClientOnCopyOnWriteStorage.java#L1227)
This test case is to check Deltastreamer when `op` is `INSERT_OVERWRITE`. Moreover using the same `SparkRDDWriteClient`. I feel it would be redundant.

Reply:
:) I get it. Guess that test also needs some fixing. 
Basically we wanna verify that insert_overwrite does not overwrite mismatched partitions. So, better to cover it as part of tests. If you feel, it might take lot of time to get this in, its ok. Do file a ticket. One of us from the community will follow up.

Reply:
created a ticket 
https://issues.apache.org/jira/browse/HUDI-2188